src/riak_kv_ts_util.erl

```diff
@@ -705,6 +708,67 @@ check_table_feature_supported(DDLRecCap, DecodedReq) ->
     end.
 
 %%
+convert_ddl_to_cluster_supported_version(DDLRecCap, DDL) when is_atom(DDLRecCap) ->
+    DDLConversions = riak_ql_ddl:convert(DDLRecCap, DDL),
+    [LowestDDL|_] = lists:sort(fun ddl_comparator/2, DDLConversions),
+    LowestDDL.
+
+ddl_comparator(A, B) ->
+    riak_ql_ddl:is_version_greater(element(1,A), element(1,B)) == true.
+
+-spec create_table(module(), ?DDL{}, proplists:proplist()) -> ok|{error,term()}.
+create_table(SvcMod, ?DDL{table = Table}=DDL1, WithProps) ->
+    DDLRecCap = riak_core_capability:get({riak_kv, riak_ql_ddl_rec_version}),
+    DDL2 = convert_ddl_to_cluster_supported_version(DDLRecCap, DDL1),
+    CompilerVersion = riak_ql_ddl_compiler:get_compiler_version(),
+    {ok, Props1} = riak_kv_ts_util:apply_timeseries_bucket_props(DDL2,
+                                                                 CompilerVersion,
+                                                                 WithProps),
+    case catch [riak_kv_wm_utils:erlify_bucket_prop(P) || P <- Props1] of
+        {bad_linkfun_modfun, {M, F}} ->
+            {error, SvcMod:make_table_create_fail_resp(Table,
+                                         flat_format(
+                                           "Invalid link mod or fun in bucket type properties: ~p:~p\n", [M, F]))};
+        {bad_linkfun_bkey, {B, K}} ->
+            {error, SvcMod:make_table_create_fail_resp(Table,
+                                         flat_format(
+                                           "Malformed bucket/key for anon link fun in bucket type properties: ~p/~p\n", [B, K]))};
+        {bad_chash_keyfun, {M, F}} ->
+            {error, SvcMod:make_table_create_fail_resp(Table,
+                                         flat_format(
+                                           "Invalid chash mod or fun in bucket type properties: ~p:~p\n", [M, F]))};
+        Props2 ->
+            create_table1(SvcMod, Table, Props2, DDLRecCap, ?TABLE_ACTIVATE_WAIT)
+    end.
+
+create_table1(SvcMod, Table, Props, DDLRecCap, Seconds) ->
+    case riak_core_bucket_type:create(Table, Props) of
+        ok -> wait_until_active_and_supported(SvcMod, Table, DDLRecCap, Seconds);
+        {error, Reason} -> {error, SvcMod:make_table_create_fail_resp(Table, Reason)}
+    end.
+
+wait_until_supported(SvcMod, Table, _DDLRecCap, _Seconds=0) ->
```

Comment:
As I understand it, this function waits for a required capabilities to be met before replying `ok`, after the table has been activated. I think this is the wrong way around. The table should be validated that it's requirements are met before it can be created, I think this was part of the table existence checks, or we can make it that way. There is no point in adding it to the metadata even if it cannot be used.

Capabilities will change when nodes come in and out of the cluster, that is unlikely to happen within a 30 second timeout. If un-negotiated capabilities is the cause of the sql log regression failures, there is a util function in rt that verifies capability values.